src/cpp/src/icontinuous_batching.cpp

```diff
@@ -110,6 +110,10 @@ ContinuousBatchingPipeline::IContinuousBatchingPipeline::generate(
         });
     }
 
+    // if streaming was canceled, prompt/answer of current step shouldn't be presented in history, so let's remove prompt from history
+    if (m_is_chat_conversation && !encoded.empty() && encoded[0].m_status == ov::genai::GenerationStatus::CANCEL)
```

Comment:
why do we need `!encoded.empty()` ? I suppose `encoded` always has the same size as number of inputs